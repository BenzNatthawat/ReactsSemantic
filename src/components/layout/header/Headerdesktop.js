import React, { Component } from 'react'
import { Container, Menu, Responsive, Segment, Visibility, Header } from 'semantic-ui-react'
import { NavLink } from 'react-router-dom'
import LoginRegister from './LoginRegister'

export class Headerdesktop extends Component {

  state = {}
  hideFixedMenu = () => this.setState({ fixed: false })
  showFixedMenu = () => this.setState({ fixed: true })

  render() {
    const { fixed } = this.state
    const origin = window.location.origin
    return (
      <Responsive minWidth={Responsive.onlyTablet.minWidth}>
        <Visibility
          once={false}
          onBottomPassed={this.showFixedMenu}
          onBottomPassedReverse={this.hideFixedMenu}
        >
          <Segment
            inverted
            textAlign='center'
            style={{ minHeight: 700, padding: '1em 0em' }}
            vertical
          >
            <Menu
              fixed={fixed ? 'top' : null}
              inverted={!fixed}
              pointing={!fixed}
              secondary={!fixed}
              size='large'
            >
              <Container>
                <Menu.Item as={NavLink} to='/' exact={true}>
                  Home
                </Menu.Item>
                <Menu.Item as={NavLink} to='/about'>Abouts</Menu.Item>
                <Menu.Item as='a'>Company</Menu.Item>
                <Menu.Item as='a'>Careers</Menu.Item>
                <LoginRegister fixed={fixed}/>
              </Container>
            </Menu>

            <Container text>
              <Header
                as='h1'
                content={origin}
                inverted
              />
              <Header
                as='h2'
                content='port 80'
                inverted
              />
            </Container>

          </Segment>

        </Visibility>
      </Responsive>
    )
  }
}

export default Headerdesktop
