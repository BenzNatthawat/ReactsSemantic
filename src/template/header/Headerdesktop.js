import React, { Component } from 'react'
import { Container, Menu, Responsive, Segment, Visibility, /*Header*/ } from 'semantic-ui-react'
import LoginRegister from './LoginRegister'
import Menulist from './Menulist'
// import Sliderimg from '../header/Sliderimg'

export class Headerdesktop extends Component {

  state = {}
  hideFixedMenu = () => this.setState({ fixed: false })
  showFixedMenu = () => this.setState({ fixed: true })

  render() {
    const { fixed } = this.state
    // const { origin, pathname } = window.location
    return (
      <Responsive minWidth={Responsive.onlyTablet.minWidth}>
        <Visibility
          once={false}
          onBottomPassed={this.showFixedMenu}
          onBottomPassedReverse={this.hideFixedMenu}
        >
          <Segment
            inverted
            textAlign='center'
            style={{ minHeight: 'auto', padding: '1em 0em' }}
            vertical
          >
            <Menu
              fixed={fixed ? 'top' : null}
              inverted={!fixed}
              pointing={!fixed}
              secondary={!fixed}
              size='large'
            >
              <Container>
                <Menulist />
                <LoginRegister fixed={fixed} />
              </Container>
            </Menu>

            {/* <Container text>
              <Header
                as='h1'
                content={origin}
                inverted
              />
              <Header
                as='h2'
                content='port 80'
                inverted
              />
            </Container> */}
            {/* {(pathname==='/'?<Sliderimg />:'')} */}
          </Segment>

        </Visibility>
      </Responsive>
    )
  }
}

export default Headerdesktop
