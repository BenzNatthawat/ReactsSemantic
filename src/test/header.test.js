import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import TestRenderer from 'react-test-renderer'
import Headerdesktop from '../components/layout/header/Headerdesktop'

test('should render Header correctly', () => {
  const renderer = new ReactShallowRenderer()
  renderer.render(<Headerdesktop />)
  expect(renderer.getRenderOutput()).toMatchSnapshot()
})

test('should render Header correctly1', () => {
  const rendered  = TestRenderer.create(<Headerdesktop />)
  console.log(rendered.root)
  // expect(wrapper.find('a').text()).toBe('')
})