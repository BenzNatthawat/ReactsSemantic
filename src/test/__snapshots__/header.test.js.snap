// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<Responsive
  getWidth={[Function]}
  minWidth={768}
>
  <Visibility
    context={[Window]}
    continuous={false}
    offset={
      Array [
        0,
        0,
      ]
    }
    onBottomPassed={[Function]}
    onBottomPassedReverse={[Function]}
    once={false}
    updateOn="events"
  >
    <Segment
      inverted={true}
      style={
        Object {
          "minHeight": 700,
          "padding": "1em 0em",
        }
      }
      textAlign="center"
      vertical={true}
    >
      <Menu
        fixed={null}
        inverted={true}
        pointing={true}
        secondary={true}
        size="large"
      >
        <Container>
          <MenuItem
            as="a"
            to="/"
          >
            Home
          </MenuItem>
          <MenuItem
            as="a"
            to="/about"
          >
            Abouts
          </MenuItem>
          <MenuItem
            as="a"
          >
            Company
          </MenuItem>
          <MenuItem
            as="a"
          >
            Careers
          </MenuItem>
          <MenuItem
            position="right"
          >
            <Button
              as="a"
              inverted={true}
            >
              Log in
            </Button>
            <Button
              as="a"
              inverted={true}
              style={
                Object {
                  "marginLeft": "0.5em",
                }
              }
            >
              Sign Up
            </Button>
          </MenuItem>
        </Container>
      </Menu>
    </Segment>
  </Visibility>
</Responsive>
`;
